### DISCUSSION TITLE
Can I reference external schemas in microservices with Prisma ORM ?

### DISCUSSION THREAD

**Discussion by francislagares** [prisma/prisma#24902] [Q&A]
### Question

Let's say I want to establish relationships between models in different services. For example, if Service Buyer needs to reference a model in Service Gig how could I do it ?

As for now I get this error Type "Gig" is neither a built-in type, nor refers to another model, custom type, or enum.

Maybe I could define Gig as a type here inside Buyer schema, but I'm not sure if this is the correct way of doing it.

### Information about Prisma Schema, Client Queries and Environment (optional)

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model Buyer {
  id             String   @id @default(auto()) @map("_id") @db.ObjectId
  username       String
  email          String
  profilePicture String
  country        String
  isSeller       Boolean  @default(false)
  createdAt      DateTime

  purchasedGigs Gig?    @relation(fields: [gigId], references: [id])
  gigId         String? @db.ObjectId

  @@index([username, email])
}

```

**Comment by ludralph** [comment]
Hi @francislagares 👋 

To establish relationships between models in different services using Prisma, you need to ensure that both models are defined within the same Prisma schema. Prisma does not support cross-schema or cross-service relationships directly. However, you can define all related models within a single schema to achieve the desired relationships.
Given your example, you want the `Buyer` model to reference the `Gig` model. Here is how you can define both models within the same Prisma schema:
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model Buyer {
  id             String   @id @default(auto()) @map("_id") @db.ObjectId
  username       String
  email          String
  profilePicture String
  country        String
  isSeller       Boolean  @default(false)
  createdAt      DateTime

  purchasedGigs  Gig?     @relation(fields: [gigId], references: [id])
  gigId          String?  @db.ObjectId

  @@index([username, email])
}

model Gig {
  id          String   @id @default(auto()) @map("_id") @db.ObjectId
  title       String
  description String
  price       Float
  createdAt   DateTime
  updatedAt   DateTime?
}
```
In this schema:

- The `Buyer` model has a relation to the `Gig` model through the `purchasedGigs` field.
- The `gigId` field in the Buyer model references the id field in the Gig model.

This setup ensures that Prisma understands the relationship between the `Buyer` and `Gig` models, and you won't encounter the error "Type 'Gig' is neither a built-in type, nor refers to another model, custom type, or enum."

For more information on defining relationships in Prisma, you can refer to the [relations documentation](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
